### PR TITLE
Upload volume format check with the supported hypervisor, fail if supported hypervisor not found on zone

### LINKED ISSUE #8652: Unable to attach a raw format volume  to an instance running on KVM
ISSUE TYPE

BUG 

Component Name

Volume, 

CLOUDSTACK VERSION

Cloudstack version 4.18, 4.19 

SUMMARY

Unable to attach a raw format volume  to an instance running on KVM 

Steps to reproduce the issue

1. Upload a volume of format RAW

Screenshot 2024-02-14 at 12 18 47 PM

[tl.raw.zip](https://github.com/apache/cloudstack/files/14276105/tl.raw.zip)

Cross check the DB ,  format is RAW

```
mysql> select * from volumes where id=8 \G;;
*************************** 1. row ***************************
                        id: 8
                account_id: 2
                 domain_id: 1
                   pool_id: NULL
              last_pool_id: NULL
               instance_id: NULL
                 device_id: NULL
                      name: test2
                      uuid: c0db18a8-153a-465c-b23d-2ec148a9b0b1
                      size: 53687296
                    folder: NULL
                      path: NULL
                    pod_id: NULL
            data_center_id: 1
                iscsi_name: NULL
                   host_ip: NULL
               volume_type: DATADISK
                 pool_type: NULL
          disk_offering_id: 6
               template_id: NULL
first_snapshot_backup_uuid: NULL
               recreatable: 0
                   created: 2024-02-13 10:19:15
                  attached: NULL
                   updated: 2024-02-13 10:21:41
                   removed: NULL
                     state: Uploaded
                chain_info: NULL
              update_count: 2
                 disk_type: NULL
    vm_snapshot_chain_size: NULL
                    iso_id: NULL
            display_volume: 1
                    format: RAW
                  min_iops: NULL
                  max_iops: NULL
             hv_ss_reserve: NULL
         provisioning_type: thin
             external_uuid: NULL
             passphrase_id: NULL
            encrypt_format: NULL
```

2. Try to attach the volume to any running vm 

Currently, Cloudstack doesn't list any running vm during volume attachment

Screenshot 2024-02-14 at 12 20 21 PM

The Backend api checks for vms which are only of Hypervisor type OVM

http://10.0.33.170:8080/client/api/?zoneid=c931b7f8-6376-47b2-8082-c4bb3867e9bf&hypervisor=Ovm&account=admin&domainid=85081752-ca47-11ee-923a-1e00d900023d&state=Stopped&command=listVirtualMachines&response=json

Not sure why Cloudstack  limiting it to only OVM hypervisor.

VM running Kvm hypervisor can have disk format in raw which can be attached easily 

I was able to attach to raw disk image to a kvm 

```

virsh attach-disk 27 /root/ tl.raw vdb

[root@ol8 ~]# virsh domblklist 27

 Target   Source
------------------------------------------------------------------------------------------
 vda      /mnt/71104366-f42e-3fff-80a4-716f643d9525/2cc496ea-27f3-47d0-b680-2796ac2ace15
 vdb      /root/tl.raw
 hdc      -
```
 
 Dumpxml
 
```
     <disk type='file' device='disk'>
      <driver name='qemu' type='raw'/>
      <source file='/root/tl.raw' index='4'/>
      <backingStore/>
      <target dev='vdb' bus='virtio'/>
      <alias name='virtio-disk1'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x08' function='0x0'/>
    </disk>
    
```
    
 

Actual Behaviour

Cloudstack is not supporting the attachement of raw disk format to KVM Hypersior 

Expected Behaviour

Cloudstack should  supporting the attachment of raw disk format to KVM Hypersior

### PR DESCRIPTION
**Pull request by sureshanaparti**
### Description

This PR checks the upload volume format with the supported hypervisor,  fail if supported hypervisor not found on zone.

Fixes #8652 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### Screenshots (if appropriate):

UploadRawVolumeError

### How Has This Been Tested?

Manually tested upload RAW volume from local.

### CODE REVIEW

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Review by vladimirpetrov — APPROVED** [review]
LGTM based on manual testing. Tested with RAW, VHD and OVA volumes on KVM hypervisor and RAW, VHD and QCOW2 on VMWare.

File: server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java [hunk 1]
```diff
@@ -564,7 +567,7 @@ private boolean validateVolume(Account caller, long ownerId, Long zoneId, String
             _resourceLimitMgr.checkResourceLimit(_accountMgr.getAccount(ownerId), ResourceType.secondary_storage);
         }
 
-        sanitizeFormat(format);
+        checkFormatWithSupportedHypervisorsInZone(format, zoneId);
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java:570]: should we still call sanitizeFormat?

Comment (sureshanaparti) [reply server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java:570]: yes @DaanHoogland, it's called before. this is duplicate call. so removed

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Pull request merged**